staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/dynamic_serving_content.go

```diff
@@ -69,7 +69,8 @@ func NewDynamicServingContentFromFiles(purpose, certFile, keyFile string) (*Dyna
 			workqueue.TypedRateLimitingQueueConfig[string]{Name: fmt.Sprintf("DynamicCABundle-%s", purpose)},
 		),
 	}
-	if err := ret.loadCertKeyPair(); err != nil {
+	ctx := context.TODO()
```

Comment:
Let's not add TODOs.

I only [see](https://grep.app/search?q=NewDynamicServingContentFromFiles) internal usage of `NewDynamicServingContentFromFiles`. Let's add a `ctx` parameter to it.

Reply:
NewDynamicServingContentFromFiles is used by many other files, you mean I have to update parameter in all of the calls of others files?

Reply:
Yes. If it's really only used internally, then we can update those calls.

The situation would be different if there were many external users of this calls. I said earlier that "I only see internal usage" - looking again, I found https://grep.app/search?q=NewDynamicServingContentFromFiles&filter[path][0]=pkg/&filter[repo][0]=SUSE/kucero

But that's still not a lot. I'm in favor of extending this API.

Reply:
you prefer add an alternative API NewDynamicServingContentFromFilesWithContext?

Reply:
has to leave context.background for NewDynamicServingContentFromFiles in staging/src/k8s.io/apiserver/pkg/server/options/serving.go and staging/src/k8s.io/cloud-provider/options/webhook.go     :-( Due to func ApplyTo() , it is called by two many files

Reply:
do you have any idea?

Reply:
I think you can add `ctx` to `NewDynamicServingContentFromFiles`. Add a comment explaining that the context is used for contextual logging. Then in those in-tree calls use `context.TODO()`, with a comment about "get context from caller for proper contextual logging".

Better not use `context.Background`, because conversion of those other components won't be done.

This defers some work in those components, but at least this package here will be done.

Reply:
got it. thanks.